src/pages/presentations/building-decentralised-applications.md

```diff
@@ -0,0 +1,862 @@
+
+You'll notice there's a few things we're missing here, such as the RPC, and the provider. That is because those are created globally as part of setting up wagmi and connecting the users wallet. So they are nicely abstracted away.
+
+For reading the contract we're using a react hook from wagmi called useReadContract. Similar to the Ethers contract object, this takes the address of the contract and the abi. We also specify here the view we want to call, and the input parameters for this function.
+
+The result object it returns has lots of useful data on it, including if the result is still pending, if it has an error, the data from the call and much more.
+
+Here we're doing some conditional rendering, showing a loading state while the view call is pending. An error state if there is an error. And then returning the user's balance once we have it.
+
+One nice feature about wagmi compared to using Ethers directly, is that this view is going to automatically update whenever the state changes on the chain. With Ethers, the call we made was only accurate at that block, and if the data changed later on it would be outdated. Using `useReadContract` however will automatically update as the user's balance changes.
+
+While this code looks a bit harder to read at first, it has many more features built in that make our lives easier when integrating.
+
+---
+
+## Sending Transaction Example
+
+```javascript
+import { useWriteContract } from "wagmi";
+import erc20Abi from "./erc20-abi.json";
+
+const UNI_ADDRESS = "0x1f9840a85d5aF5bf1D1762F925BDADdC4201F984";
+const CHASE = "0x3A61da6D37493E2f248A6832F49b52Af0a6f4Fbb";
+
+function App() {
+  const { writeContract } = useWriteContract();
+
+  return (
+    <button
+      onClick={() =>
+        await writeContract({
+          abi: erc20Abi,
+          address: UNI_ADDRESS,
+          functionName: "transfer",
+          args: [CHASE, 100],
+        })
+      }
+    >
+      Send UNI to Chase
+    </button>
+  );
+}
+```
+
+???
+
+And here we have another example, this time of sending a transaction. Similar to the example before with Ethers, this is transferring some UNI tokens to another user. The hook we are using for this is `userWriteContract` which we call by passing through the details of the contract we're calling, the function, and the args. Note that for creating this transaction, we don't need the private key of the user. That's because wagmi will handle that for us in the background, it will prompt the user to sign the transaction in their connected wallet when we call the `writeContract` function. So this implementation is much more suitable for a customer facing DApp.
+
+---
+
+## Multicall
+
+- For some DApps you may need to query significant amounts of data on chain
+- With what we learned so far, this would require one RPC call per query
+- This can be slow, and also transaction heavy
```

Comment:
The wording is a bit off here. I shouldn't really say transaction. I am meaning reading views. Have updated this to be more clear. I'm not touching much on batching transactions other than just batching reads. Let me know if you think I should mention that this can be used for batching transactions also, and about the `msg.sender` issue ect.